src/main/java/edu/cornell/kfs/sys/util/FavoriteAccountLineBuilderBase.java

```diff
@@ -0,0 +1,150 @@
+package edu.cornell.kfs.sys.util;
+
+import java.util.List;
+
+import org.kuali.kfs.sys.businessobject.GeneralLedgerPendingEntrySourceDetail;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.rice.krad.util.GlobalVariables;
+import org.kuali.rice.krad.util.ObjectUtils;
+
+import edu.cornell.kfs.sys.CUKFSKeyConstants;
+import edu.cornell.kfs.sys.businessobject.FavoriteAccount;
+import edu.cornell.kfs.sys.exception.FavoriteAccountException;
+import edu.cornell.kfs.sys.service.UserFavoriteAccountService;
+import edu.cornell.kfs.sys.service.UserProcurementProfileValidationService;
+
+/**
+ * Base class for helper objects that generate accounting line instances for a given Favorite Account object.
+ *
+ * @param <E> The element type of the list that the new accounting line will be inserted into.
+ * @param <T> The actual type of the accounting line to be inserted.
+ */
+public abstract class FavoriteAccountLineBuilderBase<E extends GeneralLedgerPendingEntrySourceDetail,T extends E> {
+
+    private UserProcurementProfileValidationService userProcurementProfileValidationService;
+    private UserFavoriteAccountService userFavoriteAccountService;

```

Comment:
I noticed on another PR that we should start making services protected member variables instead of private.